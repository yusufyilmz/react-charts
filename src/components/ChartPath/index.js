
import * as React from "react";
import * as d3 from "d3";

class ChartPath extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isHovered: false
        };
    }

    onMouseOver = () => {
        this.setState({
            isHovered: true
        });
    };

    onMouseOut = () => {
        this.setState({
            isHovered: false
        });
    };

    render() {
        let { radius, color, data, width, height, type, margin, strokeColor  } = this.props;

        var d;
        if (type === 'line') {
            const h = height - 2 * margin, w = width - 2 * margin
            const x = d3.scaleLinear()
                .range([75, w - 140])
                .domain(d3.extent(data, d => d.id))
            const y = d3.scaleLinear()
                .range([h - 150, 10])
                .domain([0, d3.max(data, d => d.value)])
            d = d3.line()
                .x(d => x(d.id))
                .y(d => y(d.value))
                .curve(d3.curveCatmullRom.alpha(0.0001))
        }
        else {
            const outerRadius = this.state.isHovered ? radius * 1.1 : radius;
            const innerRadius = radius * 0.9;
            d = d3
                .arc()
                .innerRadius(innerRadius)
                .outerRadius(outerRadius)
                .padAngle(0.01)
                .cornerRadius(2);
        }

        return (
            <g>
                <path
                    stroke={strokeColor}
                    strokeWidth={2}
                    d={d(data)}
                    fill={color}
                    onMouseOver={this.onMouseOver}
                    onMouseOut={this.onMouseOut}
                >
                </path>
            </g>
        );
    }
}


export default ChartPath;