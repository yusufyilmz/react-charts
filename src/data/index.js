
export default [
    {
        title: 'REVENUE',
        totalValue: 200000,
        valueType: 'euro',
        values: [
            {
                name: 'Tablet',
                percentage: 60,
                color: '#47d147'
            },
            {
                name: 'Smartphone',
                percentage: 40,
                color: '#006600'
            }
        ],
        datas: [
            { id: 1, value: 1.1 },
            { id: 2, value: 2.2 },
            { id: 3, value: 1.8 },
            { id: 4, value: 3.2 },
            { id: 5, value: 2.5 },
            { id: 6, value: 5.2 },
            { id: 7, value: 4.6 },
            { id: 8, value: 6.3 },
            { id: 9, value: 4.8 },
        ]
    },
    {
        title: 'IMPRESSIONS',
        totalValue: 200000000,
        valueType: '',
        values: [
            {
                name: 'Tablet',
                percentage: 40,
                color: '#1ad1ff'
            },
            {
                name: 'Smartphone',
                percentage: 60,
                color: '#002699'
            }
        ],
        datas: [
            { id: 1, value: 5.2 },
            { id: 2, value: 4.8 },
            { id: 3, value: 5.0 },
            { id: 4, value: 3.8 },
            { id: 5, value: 2.3 },
            { id: 6, value: 2.8 },
            { id: 7, value: 1.5 },
            { id: 8, value: 2.5 },
            { id: 9, value: 2.0 },
        ]
    },
    {
        title: 'VISITS',
        totalValue: 600000000,
        valueType: '',
        values: [
            {
                name: 'Tablet',
                percentage: 80,
                color: '#ffcc00'
            },
            {
                name: 'Smartphone',
                percentage: 20,
                color: '#e68a00'
            }
        ],
        datas: [
            { id: 1, value: 1.1 },
            { id: 2, value: 2.2 },
            { id: 3, value: 1.8 },
            { id: 4, value: 3.2 },
            { id: 5, value: 2.5 },
            { id: 6, value: 5.2 },
            { id: 7, value: 4.6 },
            { id: 8, value: 6.3 },
            { id: 9, value: 4.8 },
        ]
    },
]

