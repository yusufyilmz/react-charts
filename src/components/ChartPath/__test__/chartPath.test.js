

import React from 'react';
import ChartPath from '../index';
import { shallow } from 'enzyme'
import data from '../../../data';

describe('CHARTPATH COMPONENT', () => {

    let props, wrapper;

    beforeEach(() => {
        props = {
            height: 400,
            width: 300,
            radius :130,
            color: 'black',
            data: data[0].datas
        }

        wrapper = shallow(
            <ChartPath {...props} />
        );
    });

    test('renders element correctly', () => {
        expect(wrapper.find('g').length).toBe(1)
    });

    test('renders element correctly', () => {
        expect(wrapper.find('path').length).toBe(1)
    });

    test('renders props correctly', () => {
        expect(wrapper.find('path').props()['fill']).toBe(props.color)
    });

    test('renders props correctly', () => {
        expect(wrapper.find('path').props()['d']).toBeTruthy()
    });

    test('renders state correctly', () => {
        wrapper.find('path').simulate('mouseover');
        expect(wrapper.state().isHovered).toEqual(true)
    });

    test('snapchat correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})
