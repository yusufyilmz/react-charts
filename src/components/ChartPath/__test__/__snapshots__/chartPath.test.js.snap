// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHARTPATH COMPONENT snapchat correctly 1`] = `
<g>
  <path
    d="MNaN,NaNLNaN,NaNZ"
    fill="black"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    strokeWidth={2}
  />
</g>
`;
